src/Games/NexusMods.Games.StardewValley/Diagnostics.cs

```diff
@@ -134,4 +134,81 @@ internal static partial class Diagnostics
             .AddValue<string>("ModVersion")
         )
         .Finish();
+
+    [DiagnosticTemplate]
+    [UsedImplicitly]
+    internal static IDiagnosticTemplate GameVersionOlderThanMinimumGameVersionTemplate = DiagnosticTemplateBuilder
+        .Start()
+        .WithId(new DiagnosticId(Source, number: 8))
+        .WithTitle("Game Version older than supported by SMAPI")
+        .WithSeverity(DiagnosticSeverity.Critical)
+        .WithSummary("The minimum supported game version of SMAPI {SMAPIVersion} is {MinimumGameVersion}")
+        .WithDetails("""
+SMAPI version {SMAPIVersion} requires the game version to be at least {MinimumGameVersion}.
+The current game version is {CurrentGameVersion}.
+
+Due to this version mismatch, the game will **crash** on startup.
+You can solve this issue by either updating your game or downgrading SMAPI.
+
+The last supported SMAPI version for game version {CurrentGameVersion} is {LastSupportedSMAPIVersionForCurrentGameVersion}.
+You can download this SMAPI version from {SMAPINexusModsLink}.
+""")
+        .WithMessageData(messageBuilder => messageBuilder
+            .AddDataReference<ModReference>("SMAPIMod")
+            .AddValue<string>("SMAPIVersion")
+            .AddValue<string>("MinimumGameVersion")
+            .AddValue<string>("CurrentGameVersion")
+            .AddValue<string>("LastSupportedSMAPIVersionForCurrentGameVersion")
+            .AddValue<NamedLink>("SMAPINexusModsLink")
+        )
+        .Finish();
+
+    [DiagnosticTemplate]
+    [UsedImplicitly]
+    internal static IDiagnosticTemplate GameVersionNewerThanMaximumGameVersionTemplate = DiagnosticTemplateBuilder
+        .Start()
+        .WithId(new DiagnosticId(Source, number: 9))
+        .WithTitle("Game Version newer than supported by SMAPI")
+        .WithSeverity(DiagnosticSeverity.Critical)
+        .WithSummary("The maximum supported game version of SMAPI {SMAPIVersion} is {MaximumGameVersion}")
+        .WithDetails("""
+SMAPI version {SMAPIVersion} requires the game version to be lower than {MaximumGameVersion}.
+The current game version is {CurrentGameVersion}.
+
+Due to this version mismatch, the game will **crash** on startup.
+You can solve this issue by either downgrading your game to {MaximumGameVersion} or updating SMAPI.
+
+The newest supported SMAPI version for game version {CurrentGameVersion} is {NewestSupportedSMAPIVersionForCurrentGameVersion}.
```

Comment:
Same as above
`... for game version {CurrentGameVersion} is {NewestSupportedSMAPIVersionForCurrentGameVersion}.NewestSupportedSMAPIVersionForCurrentGameVersion}.`
-> `... for game version 1.6.0 is SMAPI v4.0.3`

Since the message is showing two version next to each other this helps distinguish them